mediaproviders/src/main/java/com/kaltura/playkit/providers/ott/PhoenixPlaylistProvider.java

```diff
@@ -377,8 +377,9 @@ private PKPlaylist getPKPlaylist(String playlistKs, List<KalturaMediaAsset> entr
 
             int listIndex = 0;
             for (KalturaMediaAsset kalturaMediaEntry : entriesList) {
-                if (kalturaMediaEntry == null || kalturaMediaEntry.getMediaFiles() == null || kalturaMediaEntry.getMediaFiles().isEmpty() ||
-                        kalturaMediaEntry.getImages() == null || kalturaMediaEntry.getImages().isEmpty()) {
+                String thumbnailUrl = (kalturaMediaEntry.getImages() != null && !kalturaMediaEntry.getImages().isEmpty()) ? kalturaMediaEntry.getImages().get(0).getUrl() : "";
```

Comment:
@GouravSna 
please do the URL logic only after checkin the null checks

Reply:
Done. Fixed it.